msu/files.py

```diff
@@ -1,16 +1,43 @@
 from urllib import parse
-from flask import current_app
 import boto3
 
+from flask import (
+    current_app,
+    flash,
+    request
+)
+
+
 s3_client = boto3.client('s3')
 
 def _url(bucket, loc_constraint, name):
     return 'https://%s.s3.%s.amazonaws.com/%s' % \
         (bucket, loc_constraint, name)
 
-def upload(key, data):
-    bucket = current_app.config['S3_BUCKET']
+def _upload(key, data, bucket_extension):
```

Comment:
It's better to provide the bucket directly instead of a "bucket_extension". Though it repeats code slightly more, the function is more general. Also, this function depends on the flask configuration, while all it does is upload a file.

Cut out the `bucket_extension` and just provide `bucket` directly.